CMakeLists.txt

```diff
@@ -4,28 +4,36 @@
 cmake_minimum_required (VERSION 3.6)
 
 # Set build type variable
-#	so RelWithDebInfo is treated as Release
+# so RelWithDebInfo is treated as Release
 if (${CMAKE_BUILD_TYPE} MATCHES "Debug")
-	set(AD_BUILD_TYPE "Debug")
+    set(AD_BUILD_TYPE "Debug")
 else ()
-	set(AD_BUILD_TYPE "Release")
+    set(AD_BUILD_TYPE "Release")
 endif ()
 
+# $<1:> always resolves to empty string, but the fact, that a generator
+# expression is used, forces Visual Studio generator to not create
+# configuration-specific folders for outputs, thus making output paths
+# consistent between different generators.
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY $<1:>)
```

Comment:
via @awf:  Add comment here, linking to CMake docs as in pr.

Reply:
(Given that there's a comment on this version I suppose a link to the CMake docs will suffice)

Reply:
Done (amended the last commit, diff here didn't update for me, but the link is there now).

Reply:
That looks good to me.  Probably best to wait for final @awf approval.

Reply:
Got that approval below.